### ISSUE TITLE
-A should override -W when it appears later on the command line

### ISSUE THREAD

**Issue by zackw** [rust-lang/rust-clippy#4091]
When more than one -A, -D, or -W option in a clippy invocation refers to the same lint, either directly or indirectly, whichever one came last on the command line should win.  The most important reason for this is so you can enable a category of warnings that's off by default, but then turn some of them back off again.  For example, suppose `src/demo.rs` is this:

```rust
pub fn demo() -> &'static str {
    "παράδειγμα"
}
```

then this invocation should _not_ complain about the non-ASCII characters in the string literal, but currently it does:

```
$ cargo clippy -- -W clippy::pedantic -A clippy::non-ascii-literal
warning: literal non-ASCII character detected
 --> src/demo.rs:2:5
  |
2 |     "παράδειγμα"
  |     ^^^^^^^^^^^^
  |
  = note: `-W clippy::non-ascii-literal` implied by `-W clippy::pedantic`
...
```

One might also want to be able to do `-D warnings -A clippy::something_specific` or `-W clippy::all -D clippy::particular_bad_thing`.

Version information:

```
$ cargo clippy -V
clippy 0.0.212 (726176e 2019-04-18)
```

**Comment by flip1995** [comment]
Does this work with rustc lints? E.g:
```bash
cargo check -- -D warnings -A unused
```

```rust
// main.rs
fn main() {
    let x = 42;
}
```

Otherwise this would be a rustc issue, since the flags are handled by rustc and just forwarded to Clippy.

**Comment by MikailBag** [comment]
I have same problem.
This code:
```rust
Command::new("cargo")
        .args(&[
            "clippy",
            "--all",
            "--tests",
            "--",
            "-Dclippy::all",
            "-Aclippy::new-without-default",
            "-Dwarnings",
        ])
```

does not prevent Clippy from giving me error:
```
error: you should consider adding a `Default` implementation for `global::GlobalState`

  --> src/frontend-engine/src/global.rs:15:5
...
   = note: `-D clippy::new-without-default` implied by `-D clippy::all`
```

As for rustc lints, overriding seems to work fine:
![image](https://user-images.githubusercontent.com/11889433/69449199-68416780-0d6b-11ea-81dd-b34104c7167f.png)

**Comment by casey** [comment]
A workaround is to use annotations, where a specific `allow` will override a more general `deny`. For example:

```rust
#![deny(clippy::pedantic)]
#![allow(clippy::enum_glob_use)]
```
